app/views/admin/index.html.erb

```diff
@@ -0,0 +1,42 @@
+<% origin = '' %>
+
+<h1>Requests</h1>
+<div class="admin-requests">
+
+<% for request in @requests %>
+
+  <% if origin != request.origin %>
+    <% if origin != '' %>
+      </tbody></table>
```

Comment:
I don't quite understand what's happening here.  Is this grouping the requests by origin into separate tables?